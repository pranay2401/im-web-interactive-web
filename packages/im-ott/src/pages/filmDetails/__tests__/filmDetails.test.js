import React from "react";
import { cleanup } from "@testing-library/react";
import FilmDetails from "../FilmDetails";
import renderer from "react-test-renderer";
import Router from "react-router-dom";

afterEach(cleanup);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  Link: "Link",
  useParams: jest.fn(),
  useLocation: jest.fn().mockReturnValue({
    pathname: "/film/abc-def-456-rfg-sfg",
    search: "",
    hash: "",
    state: null,
    key: "5nvxpbdafa",
  }),
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

it("FilmDetails should render correctly", () => {
  jest.spyOn(Router, "useParams").mockReturnValue({ filmId: "5678" });
  const tree = renderer.create(<FilmDetails />).toJSON();
  expect(tree).toMatchSnapshot();
});
