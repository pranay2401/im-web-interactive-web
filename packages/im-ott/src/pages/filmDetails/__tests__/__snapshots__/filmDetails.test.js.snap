// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetails should render correctly 1`] = `
<div
  className="film-details container"
>
  <div
    className="ant-skeleton ant-skeleton-element skeletonBanner"
  >
    <div
      className="ant-skeleton-image skeletonBanner"
    >
      <svg
        className="ant-skeleton-image-svg"
        viewBox="0 0 1098 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="ant-skeleton-image-path"
          d="M365.714286 329.142857q0 45.714286-32.036571 77.677714t-77.677714 32.036571-77.677714-32.036571-32.036571-77.677714 32.036571-77.677714 77.677714-32.036571 77.677714 32.036571 32.036571 77.677714zM950.857143 548.571429l0 256-804.571429 0 0-109.714286 182.857143-182.857143 91.428571 91.428571 292.571429-292.571429zM1005.714286 146.285714l-914.285714 0q-7.460571 0-12.873143 5.412571t-5.412571 12.873143l0 694.857143q0 7.460571 5.412571 12.873143t12.873143 5.412571l914.285714 0q7.460571 0 12.873143-5.412571t5.412571-12.873143l0-694.857143q0-7.460571-5.412571-12.873143t-12.873143-5.412571zM1097.142857 164.571429l0 694.857143q0 37.741714-26.843429 64.585143t-64.585143 26.843429l-914.285714 0q-37.741714 0-64.585143-26.843429t-26.843429-64.585143l0-694.857143q0-37.741714 26.843429-64.585143t64.585143-26.843429l914.285714 0q37.741714 0 64.585143 26.843429t26.843429 64.585143z"
        />
      </svg>
    </div>
  </div>
  <div
    className="metadata"
  >
    <div
      className="titleMetadata"
    >
      <div
        className="thumbnail"
      >
        <div
          className="ant-image"
          style={
            Object {
              "height": undefined,
              "width": undefined,
            }
          }
        >
          <img
            alt="undefined cover"
            className="ant-image-img"
            onError={[Function]}
            onLoad={[Function]}
            src=""
            style={
              Object {
                "height": undefined,
              }
            }
          />
        </div>
      </div>
      <div
        className="titleMargin"
      >
        <div
          className="ant-skeleton ant-skeleton-active"
        >
          <div
            className="ant-skeleton-content"
          >
            <h3
              className="ant-skeleton-title"
              style={
                Object {
                  "width": "38%",
                }
              }
            />
            <ul
              className="ant-skeleton-paragraph"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="description"
  >
    <div
      className="descriptionBorder"
    >
      <div
        className="ant-skeleton ant-skeleton-active"
      >
        <div
          className="ant-skeleton-content"
        >
          <h3
            className="ant-skeleton-title"
            style={
              Object {
                "width": "38%",
              }
            }
          />
          <ul
            className="ant-skeleton-paragraph"
          />
        </div>
      </div>
      <div
        className="ant-skeleton ant-skeleton-active"
      >
        <div
          className="ant-skeleton-content"
        >
          <h3
            className="ant-skeleton-title"
            style={
              Object {
                "width": "38%",
              }
            }
          />
          <ul
            className="ant-skeleton-paragraph"
          >
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": "61%",
                }
              }
            />
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    className="comments"
  >
    <div
      className="commentsBorder"
    >
      <div
        className="ant-skeleton ant-skeleton-active"
      >
        <div
          className="ant-skeleton-content"
        >
          <h3
            className="ant-skeleton-title"
            style={
              Object {
                "width": "38%",
              }
            }
          />
          <ul
            className="ant-skeleton-paragraph"
          />
        </div>
      </div>
      <div
        className="ant-skeleton ant-skeleton-active"
      >
        <div
          className="ant-skeleton-content"
        >
          <h3
            className="ant-skeleton-title"
            style={
              Object {
                "width": "38%",
              }
            }
          />
          <ul
            className="ant-skeleton-paragraph"
          >
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": "61%",
                }
              }
            />
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container gradientBackground"
  >
    <div
      className="ant-skeleton ant-skeleton-active"
    >
      <div
        className="ant-skeleton-content"
      >
        <h3
          className="ant-skeleton-title"
          style={
            Object {
              "width": "38%",
            }
          }
        />
        <ul
          className="ant-skeleton-paragraph"
        />
      </div>
    </div>
    <div
      className="menuContainer menuContainerFeatured"
    >
      <div
        className="filmArrow"
      >
        <span
          aria-label="left"
          className="anticon anticon-left"
          onClick={[Function]}
          role="img"
          style={
            Object {
              "color": "#000",
              "display": "block",
              "fontSize": "2.4rem",
              "height": "4rem",
              "twoToneColor": "#fff",
              "width": "4rem",
            }
          }
          tabIndex={-1}
        >
          <svg
            aria-hidden="true"
            data-icon="left"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
            />
          </svg>
        </span>
      </div>
      <div
        className="filmArrow"
      >
        <span
          aria-label="right"
          className="anticon anticon-right"
          onClick={[Function]}
          role="img"
          style={
            Object {
              "color": "#000",
              "display": "block",
              "fontSize": "2.4rem",
              "height": "4rem",
              "twoToneColor": "#fff",
              "width": "4rem",
            }
          }
          tabIndex={-1}
        >
          <svg
            aria-hidden="true"
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
